### PR TITLE
refactor: Improve select structure for accessibility and reusability 

### LINKED ISSUE #756: Feature: improve custom select usage
### Type of feature

🍕 Feature

### Current behavior

our custom select using [radix select](https://www.radix-ui.com/docs/primitives/components/select) functions like a [radix drop-down](https://www.radix-ui.com/docs/primitives/components/dropdown-menu) right now

### Suggested solution

implement the full extent of select components including arrows, scroll, tabbing, and improve the css design

### Additional context

https://github.com/open-sauced/insights/pull/705#pullrequestreview-1232548680

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Contributing Docs

- [X] I agree to follow this project's Contribution Docs

### PR DESCRIPTION
**Pull request by OgDev-01**
## What type of PR is this? (check all applicable)

- [ ] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [x] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Description
This PR: 

- Updates the previous select implementation and improves accessibility and reusability
- updates all usage of the select component
- updates selects default styles to match figma design system

## Related Tickets & Documents
Closes #756 

## Mobile & Desktop Screenshots/Recordings

image

image

image

image

image

## Added tests?

- [ ] 👍 yes
- [x] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [x] 📕 storybook
- [ ] 🙅 no documentation needed

### CODE REVIEW

**Comment by OgDev-01** [comment]
How do I add Gifs to this PR ?🥲🥲... took a whole lot to get this refactor done😄

**Comment by bdougie** [comment]
> How do I add Gifs to this PR ?🥲🥲... took a whole lot to get this refactor done😄

@BekahHW covered this in her latest post https://dev.to/opensauced/how-to-create-a-good-pull-request-template-and-why-you-should-add-gifs-4i0l

But if you are on chrome you can add [this extension](https://chrome.google.com/webstore/detail/gifs-for-github/dkgjnpbipbdaoaadbdhpiokaemhlphep?hl=en)

**Comment by OgDev-01** [comment]
Thanks

**Comment by 0-vortex** [comment]
the language selector does not have curson pointer, therefore, not hinting it is actually clickable :D

File: components/organisms/Contributors/contributors.tsx [hunk 1]
```diff
@@ -4,12 +4,15 @@ import useStore from "lib/store";
...
-import Select from "components/atoms/Select/custom-select";
 
 import { calcDistanceFromToday } from "lib/utils/date-utils";
+import color from "lib/utils/color.json";
 
 import ContributorCard from "../ContributorCard/contributor-card";
 import SkeletonWrapper from "components/atoms/SkeletonLoader/skeleton-wrapper";
+import LimitSelect from "components/atoms/Select/limit-select";
+
+const colorKeys = Object.keys(color);
```
Comment (brandonroberts) [inline components/organisms/Contributors/contributors.tsx:15]: Why was this added?

File: components/organisms/UserSettingsPage/user-settings-page.tsx [hunk 2]
```diff
@@ -20,6 +19,7 @@ import { useFetchUser } from "lib/hooks/useFetchUser";
 import { getInterestOptions } from "lib/utils/getInterestOptions";
 import { useToast } from "lib/hooks/useToast";
 import { validateTwitterUsername } from "lib/utils/validate-twitter-username";
+import { RiArrowDownSLine, RiArrowUpSLine } from "react-icons/ri";
```
Comment (brandonroberts) [inline components/organisms/UserSettingsPage/user-settings-page.tsx:22]: nit: move to top with other third-party imports

File: components/organisms/Contributors/contributors.tsx (hunk 1 above)
Comment (OgDev-01) [reply components/organisms/Contributors/contributors.tsx]: Nice catch!... My mistake...

File: stories/atoms/limit-select.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+  </div>
+);
+
+export const Default = SelectTemplate.bind({});
+
+Default.args = {
+  options: [
+    { name: "select", value: "select" },
+    { name: "food", value: "food" },
+    { name: "fruit", value: "fruit" }
```
Comment (Deadreyo) [inline stories/atoms/limit-select.stories.tsx:22]: This doesn't work. `selected` state becomes `NaN` when selecting any of them.

**Review by Deadreyo — CHANGES_REQUESTED** [review]
Limit Select story malfunctions in the design system. If you try to select an option, it won't work as `selected` state in limit story component becomes `NaN`, due to trying to convert a string of letters into a number. You may need to re-consider the type of values going in and if you want the `value` of options to be numbers or strings (strings make more sense, even if numerical strings like "10" instead of 10).

**Review by brandonroberts — APPROVED** [review]

**Pull request merged**